skyportal/skyportal
# Migrations testing

The workflow is only run when a change is detected in the `alembic/versions` folder and/or a change in `models.py` (thus should detect when someone forgets to add a migration at all when it's needed). 

Steps for testing:
- Check out latest master 
  - Should now be the on the last database revision
- Initialized the database and stamp it as head 
  - We now have a database presumably correct and at the latest version
- Check out the PR branch
- Try running the migration 
  - If no migration exists (i.e. schema changed without migration being added), this will do nothing
  - Will catch actual errors in the migration like duplicate columns, not respecting null constraints, etc.
- Try autogenerating a new migration script after the migration is run. 
  - This checks that all schema changes in the PR are actually covered in the new migration script. 
  - If there are no missing migration steps detected, the generated migration will just have `pass` blocks in the `upgrade`/`downgrade` functions, so we check for this as the success condition.
  - This generated migration script is uploaded as an artifact so that it can be reviewed for debugging.

Closes #1662
